src/Controls/StreetControls.js

```diff
@@ -77,8 +79,143 @@ class StreetControls extends FirstPersonControls {
+        // specific method to pick on the elevation layer at the good elevation, even if elevation layer is done in the shader
+        const mouse = this.view.eventToViewCoords(event);
+        const target = new THREE.Vector3();
+        if (this.view.getPickingPositionFromDepth(mouse, target)) {
+            // other way to pick object, with the basic threeJs raycast, and then objects are filtered.
+            const firstPicked = this.firstPickedObject(event);
+
+            // nothing is picked
+            if (!firstPicked) {
+                this.hideSurfaces();
+                return;
+            }
+
+            // get distance from the elevation layer
+            const distanceToGround = this.camera.position.distanceTo(target);
+            // compare first picked object distance to the distance of the elevation layer
+            if (!firstPicked.object.isTileMesh && firstPicked.distance < distanceToGround) {
+                // here we pick something, for example a building
+                target.copy(firstPicked.point);
+                // update recangle surface helper
+                this.updateSurface(target, firstPicked.face.normal, this._rectangleMesh, this._circleMesh);
+                // callback
+                pickObject(target);
+            } else {
+                // here we pick the ground, update circle surface helper
+                this.updateSurface(target, firstPicked.face.normal, this._circleMesh, this._rectangleMesh);
+                // callback
+                pickGround(target);
+            }
+        }
+    }
+
+    onClick() {
+        // here we detect mouse click when mouse is not moving,
+        // to avoid interfering with mouse drag movement, managed by FirstPersonControls
+        this.maybeClickWithoutMoving = true;
+    }
+
+    onClickWithoutMoving(event) {
+        this.pick(event, this.onClickOnGround.bind(this), this.onClickOnWall.bind(this));
+    }
+
+    onMouseUp(e) {
+        super.onMouseUp();
+
+        if (this.maybeClickWithoutMoving) {
+            this.onClickWithoutMoving(e);
+        }
+    }
+
+    onMouseMove(event) {
+        super.onMouseMove(event);
+
+        if (!this._isMouseDown) {
+            this.pick(event);
+        } else {
+            this.maybeClickWithoutMoving = false;
+        }
+    }
+
+    // create surfaces, a circle and a rectangle used to help end user
+    initSurfaces() {
+        const material = new THREE.MeshBasicMaterial({
+            side: THREE.DoubleSide,
+            color: 0xffffff,
+            depthTest: false,
+            transparent: true,
+            opacity: 0.5 });
+
+        const geomPlane = new THREE.PlaneGeometry(4, 2, 1);
+        this._rectangleMesh = new THREE.Mesh(geomPlane, material);
+        const geomCircle = new THREE.CircleGeometry(1, 32);
+        this._circleMesh = new THREE.Mesh(geomCircle, material);
+
+        this.view.scene.add(this._rectangleMesh);
```

Comment:
Why adding it if you add it below in `this.surfacesNode`, which is added to the scene ?

Reply:
yep it's changed